packages/react-api/src/hooks/useGeoJsonFeatures.js

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useCallback, useState, useRef } from 'react';
+import { useDispatch } from 'react-redux';
+import { setViewportFeaturesReady } from '@carto/react-redux';
+import { debounce } from '@carto/react-core';
+import { Methods, executeTask } from '@carto/react-workers';
+import { throwError } from './utils';
+
+export default function useGeoJsonFeatures({
+  source,
+  viewport,
+  uniqueIdProperty = 'cartodb_id',
+  debounceTimeout = 250
+}) {
+  const dispatch = useDispatch();
+  const [isGeoJsonLoaded, setGeoJsonLoaded] = useState(false);
+  const debounceIdRef = useRef(null);
+
+  const clearDebounce = () => {
+    if (debounceIdRef.current) {
+      clearTimeout(debounceIdRef.current);
+    }
+    debounceIdRef.current = null;
+  };
+
+  const sourceId = source?.id;
+
+  const setSourceViewportFeaturesReady = useCallback(
+    (ready) => {
+      if (sourceId) {
+        dispatch(setViewportFeaturesReady({ sourceId, ready }));
+      }
+    },
+    [dispatch, sourceId]
+  );
+
+  const computeFeaturesGeoJson = useCallback(
+    ({ viewport, uniqueIdProperty }) => {
+      executeTask(sourceId, Methods.VIEWPORT_FEATURES_GEOJSON, {
+        viewport: Float32Array.of(...viewport),
```

Comment:
I would remove the `Float32Array` to have a code more readable

Reply:
Agree

Reply:
Done.